lib/ansible/modules/web_infrastructure/ansible_tower/tower_inventory.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/python
+#coding: utf-8 -*-
+
+# (c) 2017, Wayne Witzel III <wayne@riotousliving.com>
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: tower_inventory
+version_added: "2.3"
+short_description: create, update, or destroy Ansible Tower inventory.
+description:
+    - Create, update, or destroy Ansible Tower inventories. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - The name to use for the inventory.
+      required: True
+    description:
+      description:
+        - The description to use for the inventory.
+      required: False
+      default: null
+    organization:
+      description:
+        - Organization the inventory belongs to.
+      required: True
+    variables:
+      description:
+        - Inventory variables. Use '@' to get from file.
```

Comment:
this is a bad pattern for a module, change to detect dict or string (assume path)